### DISCUSSION TITLE
How do I launch a program on a specific side of a workspace on startup?

### DISCUSSION THREAD

**Discussion by datajunkie1** [i3/i3#5666] [Q&A]
I've got this at the bottom of `~/.config/i3/config`:

    exec --no-startup-id i3-msg 'workspace $ws1; exec i3-sensible-terminal ranger; exec i3-sensible-terminal btop'

But when I start i3wm, it's a coinflip whether `ranger` will open on the left or right of `btop`. I want `btop` to be on the left side of workspace 1 and `ranger` to be on the right.

**Comment by okraits** [comment]
You have two options:

1. Use `sleep` to add a delay before executing ranger.
2. Use layout saving and restoring: https://i3wm.org/docs/layout-saving.html
